romancal/skycell/tests/test_skycell.py

```diff
@@ -0,0 +1,280 @@
+"""Unit tests for skycell functions"""
+
+from pathlib import Path
+
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+import romancal.skycell.skymap as sc
+
+DATA_DIRECTORY = Path(__file__).parent / "data"
+
+
+def assert_allclose_lonlat(actual: np.ndarray, desired: np.ndarray, rtol=1e-7, atol=0):
+    assert_allclose(
+        np.array(actual) % 360, np.array(desired) % 360, rtol=rtol, atol=atol
+    )
+
+
+@pytest.fixture()
+def skymap_subset() -> sc.SkyMap:
+    """
+    smaller subset to allow these tests
+    to run without access to the full skymap from CRDS.
+    """
+    return sc.SkyMap(DATA_DIRECTORY / "skymap_subset.asdf")
+
+
+def test_skycell_from_name(skymap_subset):
+    skycell = sc.SkyCell.from_name("225p90x30y51", skymap=skymap_subset)
+
+    assert skycell == sc.SkyCell(107, skymap=skymap_subset)
+
+    assert skycell.data == np.void(
+        (
+            "225p90x30y51",
+            312.1369696579,
+            88.5318370582,
+            -87.13697,
+            98499.5,
+            -2300.5,
+            313.6617771309,
+            88.4950897773,
+            313.5902417923,
+            88.5714066707,
+            310.5350393365,
+            88.5674932367,
+            310.7608411964,
+            88.4913746368,
+        ),
+        dtype=[
+            ("name", "<U16"),
+            ("ra_center", "<f8"),
+            ("dec_center", "<f8"),
+            ("orientat", "<f4"),
+            ("x_tangent", "<f8"),
+            ("y_tangent", "<f8"),
+            ("ra_corn1", "<f8"),
+            ("dec_corn1", "<f8"),
+            ("ra_corn2", "<f8"),
+            ("dec_corn2", "<f8"),
+            ("ra_corn3", "<f8"),
+            ("dec_corn3", "<f8"),
+            ("ra_corn4", "<f8"),
+            ("dec_corn4", "<f8"),
+        ],
+    )
+
+    with pytest.raises(KeyError):
+        sc.SkyCell.from_name("270p65x49y70", skymap=skymap_subset)
+
+    with pytest.raises(ValueError):
+        sc.SkyCell.from_name("r274dp63x63y81", skymap=skymap_subset)
+
+    with pytest.raises(ValueError):
+        sc.SkyCell.from_name("notaskycellname", skymap=skymap_subset)
+
+
+def test_skycell_from_asn(skymap_subset):
+    skycell = sc.SkyCell.from_asn(
+        DATA_DIRECTORY / "L3_mosaic_asn.json", skymap=skymap_subset
+    )
+    assert skycell.name == "225p90x49y67"
+
+    with pytest.raises(ValueError):
+        sc.SkyCell.from_asn(
+            DATA_DIRECTORY / "L3_regtest_asn.json", skymap=skymap_subset
+        )
+
+    with pytest.raises(ValueError):
+        sc.SkyCell.from_asn(
+            DATA_DIRECTORY / "L3_skycell_mbcat_asn.json", skymap=skymap_subset
+        )
```

Comment:
These tests are checking to make sure that the skymap_subset is incomplete, basically?  I have more sympathy for the earlier tests showing that "notaname" wasn't in the subset.